### PR TITLE
Support SkippedException in deployment phase, and skip gen1/gen2 conflict cases

### PR DESCRIPTION
**Pull request by squirrelsc**

### CODE REVIEW

File: lisa/testsuite.py [hunk 1]
```diff
@@ -84,6 +84,35 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
     def name(self) -> str:
         return self.runtime_data.metadata.name
 
+    def handle_exception(
+        self, exception: Exception, log: Logger, phase: str = ""
+    ) -> None:
+        if phase:
+            phase = f"{phase} "
+        if isinstance(exception, SkippedException):
+            ...
```
Comment (glima) [inline lisa/testsuite.py]: Why the ... here?

File: lisa/testsuite.py [hunk 2]
```diff
@@ -84,6 +84,35 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
...
+            log.info(f"case skipped: {exception}")
+            log.debug("case skipped", exc_info=exception)
+            # case is skipped dynamically
+            self.set_status(TestStatus.SKIPPED, f"{phase}skipped: {exception}")
+        elif isinstance(exception, NotRunException):
+            log.info(f"case keep NOTRUN: {exception}")
+            log.debug("case NOTRUN", exc_info=exception)
+            # case is not run dynamically.
+            self.set_status(TestStatus.NOTRUN, f"{phase}notrun: {exception}")
+        elif isinstance(PassedException, NotRunException):
```
Comment (glima) [inline lisa/testsuite.py]: Do you mean exception instead of PassedException?

File: lisa/testsuite.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/testsuite.py]: good catch, will remove it.

File: lisa/testsuite.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/testsuite.py]: fix in next commits, will move it here.

**Pull request merged**